pip_review/__main__.py

```diff
@@ -223,11 +242,23 @@ def main():
     outdated = get_outdated_packages(list_args)
     if not outdated and not args.raw:
         logger.info('Everything up-to-date')
-    elif args.auto:
-        update_packages(outdated, install_args)
+        return
     elif args.raw:
         for pkg in outdated:
             logger.info('{0}=={1}'.format(pkg['name'], pkg['latest_version']))
+        return
+    elif args.preview or args.preview_only:
+        namelen, versionlen, latestlen = get_values_maxlength(outdated)
+        previewformat = '{0:<' + str(namelen) + '} {1:<' + str(versionlen) + '} {2:<' + str(latestlen) + '} {3}'
+        logger.info(previewformat.format('Package','Version','Latest','Type'))
+        logger.info('-' * namelen + ' ' + '-' * versionlen + ' ' + '-' * latestlen + ' -----')
+        for pkg in outdated:
+            logger.info(previewformat.format(pkg['name'], pkg['version'], pkg['latest_version'], pkg['latest_filetype']))
+        logger.info('-' * (namelen + versionlen + latestlen + 8))
```

Comment:
I did this to make the `logger` object belong to `main` and to make it similar to `--raw` and `--interactive` processing.

As in the summary of the first point, this defines a function that creates the whole to be displayed in preview as one string containing multiple lines, and in the processing of this `main` part, the result of that function is displayed in `logger`. Then it will be simple.

Reply:
I'm fine with there being a function called `format_preview`, which does both data extraction and table formatting. Still, I want you to *also* write a `format_table` function which does table formatting *only*. `format_preview` should then call `format_table` as the last step.

I'm OK with you skipping the tests, but that makes proper factoring even more important.

Reply:
I see.
It is possible to try to understand if it is specifically presented.

However, I didn't have this idea, and I could only write code with simple steps.
Well, it can be said that there is a lack of knowledge and experience and cultural differences.

By the way, as a result, you have recoded everything.
So this pull request is just like being rejected.
Please handle it that way.